## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :x: [more...](#9c8d1031-ff1c-4057-9938-ed34dabb0d87) |
| mint-fs.sh | :x: [more...](#f1f55fd0-ba74-46bc-8083-4b012a19c571) |
| mint-gateway-s3.sh | :x: [more...](#2a3a22ce-3a92-4bb2-872f-bb738bec295b) |
| mint-erasure.sh | :x: [more...](#6afb9343-322e-4a7f-96c8-d84ee081d7e5) |
| mint-dist-erasure.sh | :x: [more...](#fa523fc4-42a2-4d55-ae7a-f8aa9ef41d60) |
| mint-gateway-nas.sh | :x: [more...](#adf311f1-ce52-4bcf-8a7a-27779fbcdc5a) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#2f0ebacc-2c2a-4191-b3a9-c6369e1b2d0f) |
| mint-pools.sh | :x: [more...](#2f7742c0-11bb-4b13-8219-e31e1888daae) |
<a name="2f7742c0-11bb-4b13-8219-e31e1888daae"></a>
### 13921-0aa698a/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:30933
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 1 minutes and 53 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 33 seconds
(9/15) Running minio-go tests ... FAILED in 22 seconds
{
  "args": {
    "bucketName": "minio-go-test-olptsnjztolwhmqr",
    "objectName": "o0b3kbjt3aggvhi55vjic5s0grrbgg"
  },
  "duration": 1274,
  "function": "GetObject()",
  "message": "Unexpected number of Version elements in listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 97,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-rg8fi4, objectName = null, resource = /minio-java-test-rg8fi4, requestId = 16C1123D8FC37DB1, hostId = 5556934a-25cf-4a98-b7b1-de1b64bd75cf)\nrequest={method=PUT, url=http://minio-k8s3:30933/minio-java-test-rg8fi4?encryption=, headers=Host: minio-k8s3:30933\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234314Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 342\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C1123D8FC37DB1\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:43:14 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(9/15) Running minio-js tests ... FAILED in 18 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:89:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:164:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(9/15) Running minio-py tests ... FAILED in 1 minutes and 37 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-abd901d8-db68-4934-a44b-7bf9efa727c8",
    "object_name": "3a4091aa-5034-487d-9eb6-62da959ebeea"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-abd901d8-db68-4934-a44b-7bf9efa727c8, request_id: 16C1125869F45604, host_id: 5556934a-25cf-4a98-b7b1-de1b64bd75cf, bucket_name: minio-py-test-abd901d8-db68-4934-a44b-7bf9efa727c8",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1611, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 669, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-abd901d8-db68-4934-a44b-7bf9efa727c8, request_id: 16C1125869F45604, host_id: 5556934a-25cf-4a98-b7b1-de1b64bd75cf, bucket_name: minio-py-test-abd901d8-db68-4934-a44b-7bf9efa727c8\n",
  "duration": 3935
}
(9/15) Running s3cmd tests ... done in 25 seconds
(10/15) Running s3select tests ... done in 4 seconds
(11/15) Running versioning tests ... FAILED in 30 minutes and 7 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-303wqy99339dn2",
    "objectName": "testObject",
    "retentionMode": "COMPLIANCE"
  },
  "duration": 1802018,
  "function": "/mint/build/versioning/main.go:79: main.cleanupBucket",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 10 out of 15 tests successfully.
```
<a name="6afb9343-322e-4a7f-96c8-d84ee081d7e5"></a>
### 13921-0aa698a/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:32483
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 1 minutes and 50 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 19 seconds
(8/15) Running minio-dotnet tests ... done in 27 seconds
(9/15) Running minio-go tests ... FAILED in 37 seconds
{
  "args": {
    "bucketName": "minio-go-test-kbcy5lxeokofj5zi",
    "objectName": "cpd691dlqcuwvfurf399mp2wos26hv",
    "sse": {}
  },
  "duration": 13,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 29,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-om0rkt, objectName = null, resource = /minio-java-test-om0rkt, requestId = 16C1123B516EFB81, hostId = 42ada9ba-7b2a-4fc4-a7bc-3e2e4373361e)\nrequest={method=PUT, url=http://minio-k8s2:32483/minio-java-test-om0rkt?encryption=, headers=Host: minio-k8s2:32483\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234305Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 342\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C1123B516EFB81\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:43:05 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(9/15) Running minio-js tests ... done in 51 seconds
(10/15) Running minio-py tests ... done in 47 seconds
(11/15) Running s3cmd tests ... done in 23 seconds
(12/15) Running s3select tests ... done in 2 seconds
(13/15) Running versioning tests ... done in 3 minutes and 2 seconds

Executed 13 out of 15 tests successfully.
```
<a name="2f0ebacc-2c2a-4191-b3a9-c6369e1b2d0f"></a>
### 13921-0aa698a/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s4:30472
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... FAILED in 23 seconds
{
  "name": "awscli",
  "duration": 2023,
  "function": "aws --endpoint-url http://minio-k8s4:30472 s3api copy-object --bucket awscli-mint-test-bucket-18901 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-18901/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 5031940d37e7d4e4d51295d3032beaa6, got 280a62d1c1ff1836492619f06b45b044"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 25 seconds
(7/15) Running minio-dotnet tests ... done in 32 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-2ic224ek59nyjrfs",
    "objectName": "fdxshefxgnuimdsbifspz0bwc0n4d6",
    "sse": {}
  },
  "duration": 56,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 99,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-2l9l96d, objectName = null, resource = /minio-java-test-2l9l96d, requestId = 16C112313A56F7AB, hostId = 3d9d5749-f61f-44ec-a8bf-e63f48684c88)\nrequest={method=PUT, url=http://minio-k8s4:30472/minio-java-test-2l9l96d?encryption=, headers=Host: minio-k8s4:30472\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234222Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C112313A56F7AB\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:42:21 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(8/15) Running minio-js tests ... FAILED in 56 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook for \"removeObject(bucketName, objectName, removeOpts)",
  "args": "bucketName:minio-js-test-ver-presign25247b4e-d816-4e4e-a9ae-6af5dbb372dc ",
  "duration": 5,
  "alert": "datafile-1-b ",
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:89:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:164:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/15) Running minio-py tests ... done in 1 minutes and 41 seconds
(9/15) Running s3cmd tests ... FAILED in 7 seconds
{
  "name": "s3cmd",
  "duration": "5243",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-29853/s3cmd-test-object-25725 a40f5f21-bb9b-47f8-baed-a5d2b4f177a8\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id a40f5f21-bb9b-47f8-baed-a5d2b4f177a8 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: Too many failures. Giving up on '/mint/data/datafile-65-MB')"
}
(9/15) Running s3select tests ... done in 4 seconds
(10/15) Running versioning tests ... done in 3 minutes and 7 seconds

Executed 10 out of 15 tests successfully.
```
<a name="fa523fc4-42a2-4d55-ae7a-f8aa9ef41d60"></a>
### 13921-0aa698a/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s4:31368
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 52 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 26 seconds
(8/15) Running minio-dotnet tests ... done in 31 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 12 seconds
{
  "args": {
    "bucketName": "minio-go-test-ick90oo14sqm1pvb",
    "objectName": "u0uxbj4ylmcfh9zehnclkiqxgk2nkj",
    "sse": {}
  },
  "duration": 46,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 83,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-1h4f8ak, objectName = null, resource = /minio-java-test-1h4f8ak, requestId = 16C1124521D1B76D, hostId = 3806792c-2d48-4c0d-9b3f-50df61506c60)\nrequest={method=PUT, url=http://minio-k8s4:31368/minio-java-test-1h4f8ak?encryption=, headers=Host: minio-k8s4:31368\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234347Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C1124521D1B76D\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:43:47 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(9/15) Running minio-js tests ... done in 56 seconds
(10/15) Running minio-py tests ... done in 1 minutes and 13 seconds
(11/15) Running s3cmd tests ... done in 31 seconds
(12/15) Running s3select tests ... done in 3 seconds
(13/15) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 15 tests successfully.
```
<a name="adf311f1-ce52-4bcf-8a7a-27779fbcdc5a"></a>
### 13921-0aa698a/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:32745
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 38 seconds
{
  "name": "awscli",
  "duration": 1748,
  "function": "aws --endpoint-url http://minio-k8s6:32745 s3api get-bucket-lifecycle-configuration --bucket awscli-mint-test-bucket-6665\n",
  "status": "FAIL",
  "error": "An error occurred (NotImplemented) when calling the GetBucketLifecycleConfiguration operation: A header you provided implies functionality that is not implemented"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 17 seconds
(7/15) Running minio-dotnet tests ... done in 36 seconds
(8/15) Running minio-go tests ... FAILED in 25 seconds
{
  "args": {
    "bucketName": "minio-go-test-pryr556qq1xvd6w1",
    "objectName": "te3qcxit99re2fblxhhl00d4bnb9gg",
    "sse": {}
  },
  "duration": 4,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 26,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-1qspufk, objectName = null, resource = /minio-java-test-1qspufk, requestId = 16C1121AFA59FA82, hostId = 935e952a-a414-431a-9e85-7f2b87a45757)\nrequest={method=PUT, url=http://minio-k8s6:32745/minio-java-test-1qspufk?encryption=, headers=Host: minio-k8s6:32745\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234046Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C1121AFA59FA82\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:40:46 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(8/15) Running minio-js tests ... done in 39 seconds
(9/15) Running minio-py tests ... done in 46 seconds
(10/15) Running s3cmd tests ... done in 19 seconds
(11/15) Running s3select tests ... done in 1 seconds
(12/15) Running versioning tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
<a name="9c8d1031-ff1c-4057-9938-ed34dabb0d87"></a>
### 13921-0aa698a/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:32516
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 56 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 32 seconds
(9/15) Running minio-go tests ... FAILED in 55 seconds
{
  "args": {
    "bucketName": "minio-go-test-jd43342lsoqali9a",
    "objectName": "farnrxtr4jjroxkn1lb3d32vrafyeh",
    "sse": {}
  },
  "duration": 61,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 90,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-2083d64, objectName = null, resource = /minio-java-test-2083d64, requestId = 16C112376CA73901, hostId = 07a6ba3a-56b7-4eb0-94d0-fc55418e75d3)\nrequest={method=PUT, url=http://minio-k8s6:32516/minio-java-test-2083d64?encryption=, headers=Host: minio-k8s6:32516\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234248Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C112376CA73901\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:42:48 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(9/15) Running minio-js tests ... done in 56 seconds
(10/15) Running minio-py tests ... done in 1 minutes and 24 seconds
(11/15) Running s3cmd tests ... done in 26 seconds
(12/15) Running s3select tests ... done in 4 seconds
(13/15) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 15 tests successfully.
```
<a name="2a3a22ce-3a92-4bb2-872f-bb738bec295b"></a>
### 13921-0aa698a/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:30246
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 51 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 17 seconds
(8/15) Running minio-dotnet tests ... done in 27 seconds
(9/15) Running minio-go tests ... done in 34 seconds
(10/15) Running minio-java tests ... FAILED in 27 seconds
{
  "name": "minio-java",
  "function": "addUser()",
  "duration": 276,
  "status": "FAIL",
  "error": "java.lang.RuntimeException: Request failed with response: {\"Code\":\"XMinioAdminVersionMismatch\",\"Message\":\"This 'admin' API is not supported by server in 'mode-gateway-s3'\",\"Resource\":\"/minio/admin/v3/add-user\",\"RequestId\":\"\",\"HostId\":\"98a417d0-4df9-4d4d-b57d-2dd0a89c75ad\"}\n >>> [io.minio.admin.MinioAdminClient.execute(MinioAdminClient.java:190), io.minio.admin.MinioAdminClient.addUser(MinioAdminClient.java:221), TestMinioAdminClient.addUser(TestMinioAdminClient.java:47), TestMinioAdminClient.runAdminTests(TestMinioAdminClient.java:143), FunctionalTest.runTests(FunctionalTest.java:3720), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(10/15) Running minio-js tests ... done in 36 seconds
(11/15) Running minio-py tests ... done in 48 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 6 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running versioning tests ... done in 1 seconds

Executed 14 out of 15 tests successfully.
```
<a name="f1f55fd0-ba74-46bc-8083-4b012a19c571"></a>
### 13921-0aa698a/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:31676
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 53 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 19 seconds
(8/15) Running minio-dotnet tests ... done in 26 seconds
(9/15) Running minio-go tests ... FAILED in 21 seconds
{
  "args": {
    "bucketName": "minio-go-test-we0m1yzjpu09bk39",
    "objectName": "es2yqwyfwrlaf9b9crpsbzrlsh6hxp",
    "sse": {}
  },
  "duration": 6,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 32,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-3dcj7lv, objectName = null, resource = /minio-java-test-3dcj7lv, requestId = 16C1123AB558FD4A, hostId = 49b9b29f-ebe0-48ae-a669-a42255fce9c7)\nrequest={method=PUT, url=http://minio-k8s1:31676/minio-java-test-3dcj7lv?encryption=, headers=Host: minio-k8s1:31676\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.3\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20211215T234302Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20211215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16C1123AB558FD4A\nX-Xss-Protection: 1; mode=block\nDate: Wed, 15 Dec 2021 23:43:02 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.S3Base.execute(S3Base.java:440), io.minio.S3Base.executePut(S3Base.java:772), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2328), FunctionalTest.setBucketEncryption(FunctionalTest.java:3222), FunctionalTest.runBucketTests(FunctionalTest.java:3655), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(9/15) Running minio-js tests ... done in 48 seconds
(10/15) Running minio-py tests ... done in 38 seconds
(11/15) Running s3cmd tests ... done in 20 seconds
(12/15) Running s3select tests ... done in 1 seconds
(13/15) Running versioning tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/bucket-replication.go
+++ b/cmd/bucket-replication.go
@@ -41,6 +41,9 @@ import (
 	"github.com/minio/minio/pkg/madmin"
 )
 
+// ReplicationWorkerMultiplier is suggested worker multiplier if traffic exceeds replication worker capacity
+const ReplicationWorkerMultiplier = 1.5
+
 // gets replication config associated to a given bucket name.
 func getReplicationConfig(ctx context.Context, bucketName string) (rc *replication.Config, err error) {
 	if globalIsGateway {
@@ -916,8 +919,8 @@ type ReplicationPool struct {
 // NewReplicationPool creates a pool of replication workers of specified size
 func NewReplicationPool(ctx context.Context, o ObjectLayer, sz int) *ReplicationPool {
 	pool := &ReplicationPool{
-		replicaCh:          make(chan ReplicateObjectInfo, 1000),
-		replicaDeleteCh:    make(chan DeletedObjectVersionInfo, 1000),
+		replicaCh:          make(chan ReplicateObjectInfo, 100000),
+		replicaDeleteCh:    make(chan DeletedObjectVersionInfo, 100000),
 		mrfReplicaCh:       make(chan ReplicateObjectInfo, 100000),
 		mrfReplicaDeleteCh: make(chan DeletedObjectVersionInfo, 100000),
 		objLayer:           o,
@@ -999,11 +1002,11 @@ func (p *ReplicationPool) queueReplicaTask(ctx context.Context, ri ReplicateObje
 			close(p.replicaCh)
 			close(p.mrfReplicaCh)
 		})
-		return
 	case p.replicaCh <- ri:
 	case p.mrfReplicaCh <- ri:
 		// queue all overflows into the mrfReplicaCh to handle incoming pending/failed operations
 	default:
+		logger.LogOnceIf(GlobalContext, fmt.Errorf("WARNING: Replication workers could not keep up with incoming traffic - consider increasing number of replication workers with `mc admin config set api replication_workers=%d`", p.suggestedWorkers(false)), replicationSubsystem)
 	}
 }
 
@@ -1177,4 +1180,9 @@ func scheduleReplication(ctx context.Context, objInfo ObjectInfo, o ObjectLayer,
 func scheduleReplicationDelete(ctx context.Context, dv DeletedObjectVersionInfo, o ObjectLayer, sync bool) {
 	globalReplicationPool.queueReplicaDeleteTask(GlobalContext, dv)
 	globalReplicationStats.Update(dv.Bucket, 0, replication.Pending, replication.StatusType(""), replication.DeleteReplicationType)
+}
+
+// suggestedWorkers recommends an increase in number of workers to meet replication load.
+func (p *ReplicationPool) suggestedWorkers(failQueue bool) int {
+	return int(float64(p.size) * ReplicationWorkerMultiplier)
 }

--- a/cmd/bucket-replication.go
+++ b/cmd/bucket-replication.go
@@ -651,6 +651,33 @@ func replicateObject(ctx context.Context, ri ReplicateObjectInfo, objectAPI Obje
 		if rtype == replicateNone {
 			// object with same VersionID already exists, replication kicked off by
 			// PutObject might have completed
+			if objInfo.ReplicationStatus == replication.Pending || objInfo.ReplicationStatus == replication.Failed {
+				// if metadata is not updated for some reason after replication, such as 503 encountered while updating metadata - make sure
+				// to set ReplicationStatus as Completed.Note that replication Stats would have been updated despite metadata update failure.
+				z, ok := objectAPI.(*erasureServerPools)
+				if !ok {
+					return
+				}
+				// This lower level implementation is necessary to avoid write locks from CopyObject.
+				poolIdx, err := z.getPoolIdx(ctx, bucket, object, objInfo.Size)
+				if err != nil {
+					logger.LogIf(ctx, fmt.Errorf("Unable to update replication metadata for %s/%s(%s): %w", bucket, objInfo.Name, objInfo.VersionID, err))
+				} else {
+					fi := FileInfo{}
+					fi.VersionID = objInfo.VersionID
+					fi.Metadata = make(map[string]string, len(objInfo.UserDefined))
+					for k, v := range objInfo.UserDefined {
+						fi.Metadata[k] = v
+					}
+					fi.Metadata[xhttp.AmzBucketReplicationStatus] = replication.Completed.String()
+					if objInfo.UserTags != "" {
+						fi.Metadata[xhttp.AmzObjectTagging] = objInfo.UserTags
+					}
+					if err = z.serverPools[poolIdx].getHashedSet(object).updateObjectMeta(ctx, bucket, object, fi); err != nil {
+						logger.LogIf(ctx, fmt.Errorf("Unable to update replication metadata for %s/%s(%s): %w", bucket, objInfo.Name, objInfo.VersionID, err))
+					}
+				}
+			}
 			return
 		}
 	}

--- a/cmd/xl-storage-format-v2.go
+++ b/cmd/xl-storage-format-v2.go
@@ -751,7 +751,9 @@ func (z *xlMetaV2) UpdateObjectVersion(fi FileInfo) error {
 				return nil
 			}
 		case DeleteType:
-			return errMethodNotAllowed
+			if version.DeleteMarker.VersionID == uv {
+				return errMethodNotAllowed
+			}
 		}
 	}
 

